pkg/ebpf/c/tracee.bpf.c

```diff
@@ -5051,18 +5053,86 @@ static __always_inline bool skb_revalidate_data(struct __sk_buff *skb, uint8_t *
     return true;
 }
 
+static __always_inline int is_http_protocol(struct __sk_buff *skb, uint8_t *head, uint8_t *tail, u32 l7_off) {
+#define http_min_len 7  // longest http command is "DELETE "
+
+    char *http = (void *)head + l7_off;
+
+    char http_str[http_min_len];
+    __builtin_memset((void *)&http_str, 0, sizeof(char)*http_min_len);
+
+    // get first http_min_len of layer 7
+    if ( tail-head < (u64)skb->len) {
+        // non-linear skb.
+        // use helper to load data into http_str.
+        if (bpf_skb_load_bytes(skb, l7_off, http_str, http_min_len) < 0){
+            // failed loading data into http_str - return.
+            return 0;
+        }
+    }
+    else if ((void *)http + sizeof(char)*http_min_len < (void *)tail) {
+        #ifdef CORE
+            bpf_core_read((void *)&http_str, sizeof(char)*http_min_len, http);
+        #else
+            bpf_probe_read((void *)&http_str, sizeof(char)*http_min_len, http);
+        #endif
+    }
+    else {
+        // not enough bytes in layer 7 - this is not HTTP.
+        return 0;
+    }
+
+    if (has_prefix("HTTP/", http_str, 6)) {
+        return HTTP_RESPONSE;
+    }
+    else if (has_prefix("GET ", http_str, 5) || has_prefix("POST ", http_str, 6) || has_prefix("PUT ", http_str, 5) || has_prefix("DELETE ", http_str, 8) || has_prefix("HEAD ", http_str, 6)) {
+        return HTTP_REQUEST;
+    }
+
+    return 0;
+}
+
 // decide network event_id based on created net_packet_t
-static __always_inline void set_net_event_id(net_packet_t *pkt) {
+static __always_inline void set_net_event_id(net_packet_t *pkt, struct __sk_buff *skb, uint32_t l4_hdr_off) {
     enum ports {
         DNS = 53,
     };
 
+    uint8_t *head = (uint8_t *)(long)skb->data;
+    uint8_t *tail = (uint8_t *)(long)skb->data_end;
+
+    // define variables for protocol IPPROTO_TCP
+    struct tcphdr *tcp;
+    struct tcphdr tcp_header = {0};
+    u32 l7_off;
+    int http_event_id;
+
     switch (pkt->protocol) {
+    case IPPROTO_TCP:
+        tcp = (void *)head + l4_hdr_off;
+
+        // copy the entire tcphdr struct to the bpf stack, so that bitfields of this struct could be used properly.
+        #ifdef CORE
+            bpf_core_read(&tcp_header, bpf_core_type_size(struct tcphdr), tcp);
+        #else
+            bpf_probe_read(&tcp_header, sizeof(struct tcphdr), tcp);
+        #endif
```

Comment:
You should have direct access to the packet, so I believe no need to read the bytes to the stack.
See how we do this in the tc_probe:
`struct tcphdr *tcp = (void *)head + l4_hdr_off;`